Tests/SwiftNavigationTests/ObserveTests.swift

```diff
@@ -3,26 +3,31 @@ import XCTest
 
 class ObserveTests: XCTestCase {
   #if swift(>=6)
-    @MainActor
-    func testIsolation() {
-      var count = 0
-      let token = SwiftNavigation.observe {
-        count = 1
+    func testIsolation() async {
+      await MainActor.run {
+        var count = 0
+        let token = SwiftNavigation.observe {
+          count = 1
+        }
+        XCTAssertEqual(count, 1)
+        _ = token
       }
-      XCTAssertEqual(count, 1)
-      _ = token
     }
   #endif
 
-  @MainActor
-  func testTokenStorage() {
-    var count = 0
-    observe {
-      count += 1
-    }
-    observe {
-      count += 1
+  #if !os(WASI)
+    @MainActor
+    func testTokenStorage() async {
```

Comment:
The point of this test is specifically to the `NSObject` version of `observe`, and so we can just omit it from Wasm entirely.